cmd/2019_7/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"flag"
+
+	"github.com/afarbos/aoc/pkg/intcode"
+	"github.com/afarbos/aoc/pkg/mathematic"
+	"github.com/afarbos/aoc/pkg/perm"
+	"github.com/afarbos/aoc/pkg/read"
+	"github.com/afarbos/aoc/pkg/utils"
+)
+
+var flagInput string
+
+func init() {
+	utils.Init(&flagInput)
+}
+
+func findMaxThrusterSignalWithSettings(instructions []int, settingsSet map[int]struct{}) int {
+	signal := mathematic.MaxInt()
+
+	perm.Ints(settingsSet, func(settings []int) {
+		var (
+			inputs = make([](chan int), 0, len(settings))
+		)
+
+		for range settings {
+			inputs = append(inputs, make(chan int, 1))
+		}
+
+		for amplifier, phase := range settings {
+			localInstructions := make([]int, len(instructions))
+			copy(localInstructions, instructions)
+
+			go func(amplifier, nextAmplifier int) {
+				intcode.Compute(
+					localInstructions,
+					intcode.Equals,
+					inputs[amplifier],
+					inputs[nextAmplifier],
+				)
+			}(amplifier, (amplifier+1)%len(settings))
+			inputs[amplifier] <- phase
+		}
+		inputs[0] <- 0
+
+		signal = mathematic.MaxInt(signal, <-inputs[0])
+	})
+
+	return signal
+}
+
+func findMaxThrusterSignal04(instructions []int) int {
+	settingsSet := map[int]struct{}{
+		0: {},
+		1: {},
+		2: {},
+		3: {},
+		4: {},
+	}
+
+	return findMaxThrusterSignalWithSettings(instructions, settingsSet)
+}
+
+func findMaxThrusterSignal59(instructions []int) int {
+	settingsSet := map[int]struct{}{
+		5: {},
+		6: {},
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [avoid magic numbers like '6', create a named constant for it](https://app.codacy.com/manual/afarbos/aoc/pullRequest?prid=5498712)